### PR TITLE
owner, metrics: add metrics

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @lamxTyler

File: owner/manager.go [hunk 1]
```diff
@@ -171,6 +180,10 @@ func (m *ownerManager) campaignLoop(ctx goctx.Context, etcdSession *concurrency.
 	logPrefix := fmt.Sprintf("[%s] %s ownerManager %s", m.prompt, m.key, m.id)
 	var err error
 	for {
+		if err != nil {
+			metrics.CampaignOwnerHistogram.WithLabelValues(logPrefix, err.Error()).Inc()
```
Comment (shenli) [inline owner/manager.go]: Why not use counter? You just use method Inc().

Comment (zimulala) [reply owner/manager.go]: Sorry, I use the wrong name.

**Comment by zimulala** [comment]
PTAL @shenli

**Comment by shenli** [comment]
LGTM
@winkyao @coocood PTAL

File: owner/manager.go [hunk 2]
```diff
@@ -149,7 +155,10 @@ func NewSession(ctx goctx.Context, logPrefix string, etcdCli *clientv3.Client, r
 		if failedCnt%logIntervalCnt == 0 {
 			log.Warnf("%s failed to new session, err %v", logPrefix, err)
 		}
+
+		metrics.NewSessionHistogram.WithLabelValues(logPrefix, metrics.RetLabel(err)).Observe(time.Since(startTime).Seconds())
```
Comment (coocood) [inline owner/manager.go]: We can move this line before error check, move `startTime` before `NewSession` and remove `defer` function.

**Comment by zimulala** [comment]
PTAL @coocood

File: owner/manager.go [hunk 3]
```diff
@@ -171,6 +176,10 @@ func (m *ownerManager) campaignLoop(ctx goctx.Context, etcdSession *concurrency.
 	logPrefix := fmt.Sprintf("[%s] %s ownerManager %s", m.prompt, m.key, m.id)
 	var err error
 	for {
+		if err != nil {
+			metrics.CampaignOwnerCounter.WithLabelValues(logPrefix, err.Error()).Inc()
```
Comment (tiancaiamao) [inline owner/manager.go]: It seems both `logPrefix` and `err.Error()` are not good label?
Maybe we'll get too much statistic kinds but each with very small count.

File: metrics/owner.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	Cancelled         = "cancelled"
+	Deleted           = "deleted"
+	SessionDone       = "session_done"
+	CtxDone           = "context_done"
+	WatchOwnerCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "tidb",
+			Subsystem: "owner",
+			Name:      "watch_owner",
+			Help:      "Bucketed histogram of processing time (s) of watch owner.",
```
Comment (lamxTyler) [inline metrics/owner.go]: Is it not `processing time`?

File: metrics/owner.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+		}, []string{"type", "return_reason"})
+
+	NoLongerOwner        = "no_longer_owner"
+	CampaignOwnerCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "tidb",
+			Subsystem: "owner",
+			Name:      "campaign_owner",
+			Help:      "Bucketed histogram of processing time (s) of campaign owner.",
```
Comment (lamxTyler) [inline metrics/owner.go]: Ditto.

**Comment by zimulala** [comment]
PTAL @tiancaiamao @lamxTyler @coocood

**Comment by tiancaiamao** [comment]
LGTM

**Comment by tiancaiamao** [comment]
@coocood @lamxTyler

**Comment by coocood** [comment]
LGTM

**Pull request merged**